Sources/Plasma/CoreLib/HeadSpin.cpp

```diff
@@ -118,17 +130,23 @@ void ErrorAssert(int line, const char* file, const char* fmt, ...)
     va_list args;
     va_start(args, fmt);
     vsnprintf(msg, arrsize(msg), fmt, args);
-#if defined(HS_DEBUGGING) && defined(_MSC_VER)
+#if defined(HS_DEBUGGING)
+#if defined(_MSC_VER)
     if (s_GuiAsserts)
     {
-        if(_CrtDbgReport(_CRT_ASSERT, file, line, NULL, msg))
+        if (_CrtDbgReport(_CRT_ASSERT, file, line, NULL, msg))
             DebugBreak();
     } else
-#endif // HS_DEBUGGING
-      if (DebugIsDebuggerPresent()) {
+#endif // _MSC_VER
+    {
         char str[] = "-------\nASSERTION FAILED:\nFile: %s   Line: %i\nMessage: %s\n-------";
         DebugMsg(str, file, line, msg);
+        DebugBreakIfDebuggerPresent();
+
+        // In case the debug trap is ignored
+        abort();
```

Comment:
That's a good point...  Right now it would behave differently in MSVC vs Unix -- I'll revise it to allow proceeding from a debugger, but continue to abort if no debugger is attached.